src/test/scala/chiselTests/CompatibilityInteroperabilitySpec.scala

```diff
@@ -351,4 +351,43 @@ class CompatibilityInteroperabilitySpec extends ChiselFlatSpec {
     compile(new Top(true))
     compile(new Top(false))
   }
+
+  "A unidirectional but flipped Bundle with something close to NotStrict compileOptions, but not exactly" should "bulk connect in import chisel3._ code correctly" in {
+    object Compat {
+      import Chisel.{defaultCompileOptions => _, _}
+      // arbitrary thing to make this *not* exactly NotStrict
+      implicit val defaultCompileOptions = new chisel3.ExplicitCompileOptions.CompileOptionsClass(
+        connectFieldsMustMatch = false,
+        declaredTypeMustBeUnbound = false,
+        dontTryConnectionsSwapped = false,
+        dontAssumeDirectionality = false,
+        checkSynthesizable = false,
+        explicitInvalidate = false,
+        inferModuleReset = true // different from NotStrict, to ensure case class equivalence to NotStrict is false
```

Comment:
It would be good to check that comment with an assertion, ie. `assert defaultCompileOptions != ExplicitCompileOptions.NotStrict`